### PR TITLE
Enabled NNLib kernel API calls for activations on the HiFi-IQ core - nne_dev

### PR DESCRIPTION
**Pull request by unmeshna017**
Enabled NNLib kernel API calls for activations on the HiFi-IQ core.

Adding @cad-audio and @joshih-cad as reviewers.

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/logistic.cc [hunk 1]
```diff
@@ -74,7 +77,22 @@ TfLiteStatus LogisticEval(TfLiteContext* context, TfLiteNode* node) {
...
+      int8_t* output_data_ptr = tflite::micro::GetTensorData<int8_t>(output);
+
+      TF_LITE_ENSURE_EQ(
+          context,
+          xa_nn_vec_apply_lut_asym8s_asym8s(
+              output_data_ptr, input_data_ptr,
+              static_cast<int8_t*>(xtensa_data->sigmoid_lut),
+              256, flat_size),
+          0);
+#elif defined(HIFI3) || defined(HIFI4)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/logistic.cc]: We may have to remove this #else part if the xa_nn_vec_apply_lut_asym8s_asym8s API is available in HiFi4 NNLib

File: tensorflow/lite/micro/kernels/xtensa/logistic_common.cc [hunk 2]
```diff
@@ -118,9 +120,37 @@ TfLiteStatus CalculateArithmeticOpDataLogistic(TfLiteContext* context,
 
 TfLiteStatus LogisticPrepare(TfLiteContext* context, TfLiteNode* node) {
   TFLITE_DCHECK(node->user_data != nullptr);
-  OpDataLogistic* data = static_cast<OpDataLogistic*>(node->user_data);
+  auto* xtensa_data = static_cast<OpDataLogisticXtensa*>(node->user_data);
+  TF_LITE_ENSURE_OK(context, CalculateArithmeticOpDataLogistic(
+                                  context, node, &xtensa_data->reference_op_data));
 
-  return CalculateArithmeticOpDataLogistic(context, node, data);
+#if defined(HIFI5) || defined(HIFI_IQ)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/logistic_common.cc]: We may have to update this for other HiFi targets if the xa_nn_vec_apply_lut_asym8s_asym8s API is available in HiFi4 NNLib

File: tensorflow/lite/micro/kernels/xtensa/softmax.cc [hunk 3]
```diff
@@ -214,7 +218,7 @@ TfLiteStatus EvalFloat32(TfLiteContext* context, TfLiteNode* node) {
 
   TFLITE_DCHECK(node->user_data != nullptr);
 
-#if defined(HIFI3) || defined(HIFI4) || defined(HIFI5)
+#if defined(HIFI3) || defined(HIFI4) || defined(HIFI5)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/softmax.cc]: remove trailing whitespace

File: tensorflow/lite/micro/kernels/xtensa/tanh.cc [hunk 4]
```diff
@@ -172,25 +193,42 @@ TfLiteStatus TanhEval(TfLiteContext* context, TfLiteNode* node) {
...
+      output_data_ptr = tflite::micro::GetTensorData<int8_t>(output);
+
+      TF_LITE_ENSURE_EQ(
+          context,
+          xa_nn_vec_apply_lut_asym8s_asym8s(
+              output_data_ptr, input_data_ptr,
+              static_cast<int8_t*>(data->tanh_lut),
+              256, flat_size),
+          0);
+#elif defined(HIFI4) || defined(HIFI3)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/tanh.cc]: We may have to remove this #else part if the xa_nn_vec_apply_lut_asym8s_asym8s API is available in HiFi4 NNLib

**Review by joshih-cad — CHANGES_REQUESTED** [review]

**Review by joshih-cad — APPROVED** [review]

File: tensorflow/lite/micro/kernels/xtensa/logistic.cc [hunk 5]
```diff
@@ -74,35 +77,34 @@ TfLiteStatus LogisticEval(TfLiteContext* context, TfLiteNode* node) {
...
+              256, flat_size),
           0);
 #else
       reference_integer_ops::Logistic(
           data->input_zero_point, data->input_range_radius,
           data->input_multiplier, data->input_left_shift,
           NumElements(input->dims), tflite::micro::GetTensorData<int8_t>(input),
           tflite::micro::GetTensorData<int8_t>(output));
-#endif  // defined(HIFI3) || defined(HIFI4) || defined(HIFI5)
+#endif  // defined(HIFI5) || defined(HIFI_IQ)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/logistic.cc]: add HiFi3/4 defines in the comment

File: tensorflow/lite/micro/kernels/xtensa/logistic_common.cc [hunk 6]
```diff
@@ -118,9 +120,37 @@ TfLiteStatus CalculateArithmeticOpDataLogistic(TfLiteContext* context,
...
+            xtensa_data->reference_op_data.input_range_radius,
+            xtensa_data->reference_op_data.input_multiplier,
+            xtensa_data->reference_op_data.input_left_shift),
+        0);
+  } else {
+    xtensa_data->sigmoid_lut = nullptr;
+  }
+
+  micro_context->DeallocateTempTfLiteTensor(input);
+#endif  // defined(HIFI5) || defined(HIFI_IQ)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/logistic_common.cc]: add HiFi3/4 defines in the comment

File: tensorflow/lite/micro/kernels/xtensa/tanh.cc [hunk 7]
```diff
@@ -150,6 +151,26 @@ TfLiteStatus TanhPrepare(TfLiteContext* context, TfLiteNode* node) {
   data->input_zero_point = input->params.zero_point;
   TF_LITE_ENSURE_OK(context, CalculateArithmeticOpData(context, node, data));
 
+#if defined(HIFI5) || defined(HIFI_IQ)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/tanh.cc]: add HiFi3/4 defines here

File: tensorflow/lite/micro/kernels/xtensa/tanh.cc [hunk 8]
```diff
@@ -150,6 +151,26 @@ TfLiteStatus TanhPrepare(TfLiteContext* context, TfLiteNode* node) {
...
+            static_cast<int8_t*>(data->tanh_lut),
+            data->input_zero_point,
+            data->input_range_radius,
+            data->input_multiplier,
+            data->input_left_shift),
+        0);
+  } else {
+    data->tanh_lut = nullptr;
+  }
+#endif  // defined(HIFI5) || defined(HIFI_IQ)
```
Comment (joshih-cad) [inline tensorflow/lite/micro/kernels/xtensa/tanh.cc]: add HiFi3/4 defines here

**Review by joshih-cad — CHANGES_REQUESTED** [review]

**Review by joshih-cad — APPROVED** [review]

**Review by cad-audio — APPROVED** [review]
